client/blocks/subscription-length-picker/index.jsx

```diff
@@ -106,20 +138,26 @@ export class SubscriptionLengthPicker extends React.Component {
 	};
 }
 
-export function myFormatCurrency( price, code ) {
+export function myFormatCurrency( price, code, options = {} ) {
 	const precision = CURRENCIES[ code ].precision;
 	const EPSILON = Math.pow( 10, -precision ) - 0.000000001;
 
 	const hasCents = Math.abs( price % 1 ) >= EPSILON;
-	return formatCurrency( price, code, hasCents ? {} : { precision: 0 } );
+	return formatCurrency( price, code, hasCents ? options : { ...options, precision: 0 } );
 }
 
 export const mapStateToProps = ( state, { plans } ) => {
 	const selectedSiteId = getSelectedSiteId( state );
+	const paymentCountryCode = getPaymentCountryCode( state );
+	const paymentPostalCode = getPaymentPostalCode( state );
+	const taxRateRequest = requestTaxRate( paymentCountryCode, paymentPostalCode );
+	const taxRate = taxRateRequest.state === 'success' ? taxRateRequest.data : undefined;
```

Comment:
Just a minor FYI, `taxRateRequest.data` will be `undefined` if it's not there, so this can be just `const taxRate = taxRateRequest.data` if I'm not mistaken.